src/coreclr/src/jit/emitxarch.cpp

```diff
@@ -11575,6 +11575,21 @@ BYTE* emitter::emitOutputRRR(BYTE* dst, instrDesc* id)
 
     noway_assert(!id->idGCref());
 
+    if (!emitInsCanOnlyWriteSSE2OrAVXReg(id))
+    {
+        switch (id->idInsFmt())
+        {
+            case IF_RWR_RRD_RRD:
+            case IF_RWR_RRD_RRD_CNS:
+            case IF_RWR_RRD_RRD_RRD:
```

Comment:
Any particular reason why these IF, I would think the latter two don't have any which write to an integer register.

Reply:
It wasn't obvious to me how to go from an IF_xxx to all the instructions that it might represent.

Reply:
It isn't very straightforward/easy.

The various `emitIns_*` methods say the format via `id->idInsFmt` (e.g. https://github.com/dotnet/runtime/blob/master/src/coreclr/src/jit/emitxarch.cpp#L4386).

So, AFAIK, you have to reverse your way back through the calls to those methods (and in some cases back to other callers or tables, where `ins` is picked up and passed in separately; as is the case with many HWIntrinsics).

Reply:
How about I add an assert that those forms aren't expected here, because of the guarding `canOnlyWrite...`.

Reply:
I dont think there is any harm in having them here, was mostly just interested.

It might be nice if we had a better way to map these in the future :)

Reply:
Ah, ok. Will go ahead and merge then.